Sources/GenIR/DebugData.swift

```diff
@@ -0,0 +1,267 @@
+			- copy of the xcarchive folder
+		- data.zip
+*/
+struct DebugData {
+
+	let captureDebugData: Bool
+	var zipBasePath: URL
+	var zipCollectionPath: URL
+
+	init (capturePath: URL?, xcodeLogPath: URL) throws {
+		// Determine whether we should capture debug data
+		guard let zipPath = capturePath else {
+			captureDebugData = false
+			zipBasePath = URL(fileURLWithPath: "")
+			zipCollectionPath = URL(fileURLWithPath: "")
+			return
+		}
+
+		// Make sure the directory to hold debug data exists and is empty
+		if !FileManager.default.directoryExists(at: zipPath) {
+			try FileManager.default.createDirectory(at: zipPath, withIntermediateDirectories: true)
+		} else {
+			if FileManager.default.contents(atPath: zipPath.absoluteString) != nil {
+				throw ValidationError("Path \(zipPath) is not empty! The directory to capture debug data must be empty.")
+			}
+		}
+
+		// Create a subdirectory for the logs and add a file log handler to write the log there.
+		self.zipBasePath = zipPath
+		self.zipCollectionPath = zipPath.appendingPathComponent("data")
+		let zipLogPath = zipCollectionPath.appendingPathComponent("log")
+		try FileManager.default.createDirectory(at: zipLogPath, withIntermediateDirectories: true)
+
+		logger.info("Debug data will be captured to: \(zipLogPath)")
+		var captureLog = FileLogHandler(filePath: zipLogPath.filePath)
+		captureLog.logLevel = logger.logLevel
+		MultiLogHandler.addHandler(captureLog)
+
+		captureDebugData = true
+
+		try collectExecutionContext(logPath: xcodeLogPath)
+	}
+
+	private func collectExecutionContext(logPath: URL) throws {
+
+		// Collect the xcodebuild log
+		try FileManager.default.copyItem(at: logPath, to: zipCollectionPath.appendingPathComponent("xcodebuild.log"))
+
+		// Collect the configured developer directory
+		let developerDir = "DEVELOPER_DIR: " + ( try execShellCommand(command: "xcode-select", args: ["-p"]) )
+
+		// Collect a possible override of the developer directory
+		let developerDirOverride = ProcessInfo.processInfo.environment["DEVELOPER_DIR"] ?? "Not set"
+
+		// Collect the xcodebuild version
+		let xcodeBuildVersion = try execShellCommand(command: "xcodebuild", args: ["-version"])
+
+		// Collect the swift version
+		let swiftVersion = try execShellCommand(command: "swift", args: ["-version"])
+
+		do {
+				let versionsUrl = zipCollectionPath.appendingPathComponent("versions.txt")
+				try developerDir.write(to: versionsUrl, atomically: true, encoding: .utf8)
+				let versionsFile = try FileHandle(forWritingTo: versionsUrl)
+				versionsFile.seekToEndOfFile()
+				// swiftlint:disable non_optional_string_data_conversion
+				versionsFile.write("DEVELOPER_DIR_OVERRIDE: \(developerDirOverride)\n".data(using: .utf8)!)
+				versionsFile.write("XCODEBUILD_VERSION: \(xcodeBuildVersion)\n".data(using: .utf8)!)
+				versionsFile.write("SWIFT_VERSION: \(swiftVersion)\n".data(using: .utf8)!)
+				// swiftlint:enable non_optional_string_data_conversion
+				versionsFile.closeFile()
```

Comment:
This could be simplified and improved by using the existing API `Process.runShell` to get the optional `stdout` and proving a default 'not set' message, then using https://developer.apple.com/documentation/swift/stringprotocol/write(to:atomically:encoding:) to write a single string once instead of writing to a file multiple times. Swift allows for multi-line strings with `"""` and interpolation via `\(someExpression)`